src/components/[guild]/DiscordBotPermissionsChecker.tsx

```diff
@@ -0,0 +1,228 @@
+import {
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Stack,
+  Text,
+  ToastId,
+  useDisclosure,
+} from "@chakra-ui/react"
+import Button from "components/common/Button"
+import DiscordRoleVideo from "components/common/DiscordRoleVideo"
+import { Modal } from "components/common/Modal"
+import { ActionToastOptions, useToastWithButton } from "hooks/useToast"
+import { Check, Wrench } from "phosphor-react"
+import { useEffect, useRef, useState } from "react"
+import useSWRImmutable from "swr/immutable"
+import { PlatformType } from "types"
+import fetcher from "utils/fetcher"
+import useGuild from "./hooks/useGuild"
+import useGuildPermission from "./hooks/useGuildPermission"
+
+const GUILD_BOT_ROLE_NAME = "Guild.xyz bot"
+/**
+ * Mapping permission names which we get from our backend to actual permission names
+ * which the user will be able to find on Discord
+ */
+const PRETTY_PERMISSION_NAMES: Record<string, string> = {
```

Comment:
This should be handled in our Discord runner (`utils` -> `getMissingPermissions`), since we defined the permission names manually there, so I'll open a PR in that repo too!

Reply:
note: [opened a PR](https://github.com/guildxyz/discord-runner/pull/517) on this, so I'll revert this change once it's merged